hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -259,17 +264,25 @@ protected Dialect() {
 
 	protected Dialect(DatabaseVersion version) {
 		this.version = version;
+		checkVersion();
 		registerDefaultKeywords();
 		initDefaultProperties();
 	}
 
 	protected Dialect(DialectResolutionInfo info) {
 		this.version = info.makeCopy();
+		checkVersion();
 		registerDefaultKeywords();
 		registerKeywords(info);
 		initDefaultProperties();
 	}
 
+	protected void checkVersion() {
+		if(this.version != null && this.version.isBefore( getMinimumSupportedVersion() )) {
```

Comment:
Let's give people a chance to use an overridden version by calling the `getVersion` method instead.

```suggestion
		final DatabaseVersion version = getVersion();
		if (version != null && version.isBefore( getMinimumSupportedVersion() ) ) {
```

Reply:
:+1: